pyNN/nest/electrodes.py

```diff
@@ -0,0 +1,120 @@
+"""
+Definition of NativeElectrodeType class for NEST.
+"""
+
+import numpy
+import nest
+from pyNN.standardmodels import electrodes, build_translations, StandardCurrentSource
+from pyNN.common import Population, PopulationView, Assembly
+from pyNN.parameters import ParameterSpace, Sequence
+from pyNN.nest.simulator import state
+from pyNN.nest.standardmodels.electrodes import NestCurrentSource
+from pyNN.models import BaseCellType
+from pyNN.nest.cells import NativeCellType
+from pyNN.nest.cells import get_receptor_types
+from pyNN.nest.cells import get_recordables
+from . import conversion
+
+
+def get_defaults(model_name):
+    valid_types = (int, float, Sequence)
+    defaults = nest.GetDefaults(model_name)
+    variables = defaults.get('recordables', [])
+    ignore = ['archiver_length', 'available', 'Ca', 'capacity', 'elementsize',
+              'frozen', 'instantiations', 'local', 'model', 'needs_prelim_update',
+              'recordables', 'state', 't_spike', 'tau_minus', 'tau_minus_triplet',
+              'thread', 'vp', 'receptor_types', 'events', 'global_id',
+              'element_type', 'type', 'type_id', 'has_connections', 'n_synapses',
+              'thread_local_id', 'node_uses_wfr', 'supports_precise_spikes',
+              'synaptic_elements'
+                ]
+    default_params = {}
+    default_initial_values = {}
+    for name, value in defaults.items():
+        if name in variables:
+            default_initial_values[name] = value
+    return default_params, default_initial_values
+    
+# Native model non standard
+def native_electrode_type(model_name):
+    """
+    Return a new NativeElectrodeType subclass.
+    """
+    assert isinstance(model_name, str)
+    default_parameters, default_initial_values = get_defaults(model_name)  
+    receptor_types = get_receptor_types(model_name) # get_receptor_types imported from nest.cells.py
+    return type(model_name,
+               (NativeElectrodeType,),
+                {'nest_model': model_name,
+                 'default_parameters': default_parameters,
+                 'default_initial_values': default_initial_values,
+                 'injectable': ("V_m" in default_initial_values),
+                 'nest_name': {"on_grid": model_name, "off_grid": model_name},
+                })
+    
+
+# Should be usable with any NEST current generator
+class NativeElectrodeType(NestCurrentSource):
+    
+    _is_computed = True
+    _is_playable = True
+
+    def __init__(self, **parameters):
+        self._device = nest.Create(self)
+        self.cell_list = [] 
+        self.phase_given = 0.0  # required for PR #502
```

Comment:
This should be removed. We interpret phase differently for standard models, but a native model should have the same behaviour as passing the same parameters in a pure NEST simulation.